app/controllers/catalog_controller.rb

```diff
@@ -1140,6 +1140,18 @@ def service_dialog_from_ot_submit_save
     end
   end
 
+  def service_dialog_generator(template)
+    # TODO: the special cases for Amazon and Azure should be removed once the service class is moved to the provider gem
+    if template.kind_of? OrchestrationTemplateCfn
+      return OrchestrationTemplateCfnDialogService
+    elsif template.kind_of? OrchestrationTemplateAzure
+      return OrchestrationTemplateAzureDialogService
+    end
+
+    # TODO: if a template can be used by multiple manager types, the UI should provide a dropdown box for user to select
+    "#{template.class.eligible_manager_types.first}::#{template.class.name.demodulize}DialogService".constantize
```

Comment:
Do you think the same template could be displayed differently in different eligible managers? I suspect the TODO refers to this, i.e. the user will have to choose the provider first and only they will they get the complete template?